docs/content/latest/develop/best-practices.md

```diff
@@ -0,0 +1,195 @@
+Postgresql [`jsonb`](https://www.postgresql.org/docs/current/datatype-json.html) datatype. It is stored on disk in
+binary format making searching & retrieval faster.
+
+## Consistent & global Secondary indexes
+To add secondary indexes you need to create tables with transactions enabled. Indexes are global and strongly consistent
+(ACID). Indexes can also be used as materialized views by using the `INCLUDE` [clause](../../api/ycql/ddl_create_index#included-columns).
+
+## Tablet leaders in multi-az and multi-region deployments
+In multi-region deployments, we can hint the database to try and keep all tablet-leaders
+in 1 region thus lowering the latency and network hops during read/write transactions.
+That can be done using the `set_preferred_zones` subcommand of [yb-admin](../../admin/yb-admin) cli.
+
+## Settings for ci/cd/integration-tests:
+Using YugabyteDB in (ci,cd,automated tests) scenarios we can set certain gflags 
+to lower durability and increase performance.
+
+Tserver gflags:
+
+1. `--fs_data_dirs` to temp/memory directory
+2. `--yb_num_shards_per_tserver=1`
+3. `--durable_wal_write=false`
+4. `--interval_durable_wal_write_ms=10000`
+5. `--bytes_durable_wal_write_mb=25`
+
+Master gflags:
+
+1. `--yb_num_shards_per_tserver=1`
+2. `--replication_factor=1`
+
+## Single AZ deployments
+In single AZ deployments, you need to set `--durable_wal_write=true` [gflag](../../reference/configuration/yb-tserver) in 
+tserver to not lose data if the whole datacenter goes down (power failure etc).
+
+## Hard Disk Drives (HDD)
+Currently HDD aren't supported by YugabyteDB. One of the reasons is that each tablet has it's own 
+WAL (write ahead log) and even though all writes are sequential, having multiple WALs, even though each WAL is written sequentially, 
+the writes may end up as random-writes when multiple WAL are flushing at the same time. 
+We're working hard on moving on per-server WAL 
+to reduce overhead and making HDD a possible alternative. You can track 
+[1K+ tablets issue](https://github.com/yugabyte/yugabyte-db/issues/1317).
+
+## Number of tables
+Each table is split into tablets and each tablet has overhead. 
+See [tablets per server](#tablets-per-server) for limits.
+
+## Tablets per server
+Each table consists of several tablets based on the `yb_num_shards_per_tserver`
+gflag. Each `tablet` has overhead in the memory of the tserver that it resides.
+Currently we recommend a maximum of around `500` tablets on each tserver.
+You have to keep this number in mind depending on the number of tables and number 
+of tablets per-server that you intend to create.
+We're [actively working](https://github.com/yugabyte/yugabyte-db/issues/1317) to increase this limit.
+
+On deployments where we have many small tables and few big tables, we can use [colocation](../explore/colocated-tables/linux.md) in YSQL layer to group small 
+tables into 1 tablet.
+
+## Use `TRUNCATE` to empty tables instead of `DELETE`
```

Comment:
This is more applicable in the YSQL/YCQL sections than here.

Reply:
In this page I'm keeping practices that apply to both YCQL & YSQL. But agree that they should be in their specific pages.